examples/ct/cert_checker_test.go

```diff
@@ -23,154 +23,133 @@ import (
-
-	isPrecert, err := IsPrecertificate(cert)
-	if err != nil {
-		t.Fatalf("Unexpected error from precert check %v", err)
-	}
-	if !isPrecert {
-		t.Fatal("Valid precert not recognized")
-	}
-
-	// Wipe all the extensions and try again
+func wipeExtensions(cert *x509.Certificate) *x509.Certificate {
 	cert.Extensions = cert.Extensions[:0]
-	isPrecert, err = IsPrecertificate(cert)
-
-	if err != nil {
-		t.Fatalf("Unexpected error from precert check %v", err)
-	}
-	if isPrecert {
-		t.Fatal("Non precert misclassified")
-	}
-}
-
-func TestIsPrecertificateNormalCert(t *testing.T) {
-	cert := pemToCert(t, testonly.CACertPEM)
-
-	isPrecert, err := IsPrecertificate(cert)
-
-	if err != nil {
-		t.Fatalf("Unexpected error from precert check %v", err)
-	}
-	if isPrecert {
-		t.Fatal("Non precert misclassified")
-	}
-}
-
-func TestIsPrecertificateInvalidNonCriticalExtension(t *testing.T) {
-	cert := pemToCert(t, testonly.PrecertPEMValid)
-	// Invalid because it's not marked as critical
-	ext := pkix.Extension{Id: ctPoisonExtensionOID, Critical: false, Value: asn1NullBytes}
-
-	cert.Extensions = []pkix.Extension{ext}
-	_, err := IsPrecertificate(cert)
-	if err == nil {
-		t.Fatal("incorrectly accepted non critical CT extension")
-	}
-}
-
-func TestIsPrecertificateInvalidBytesInExtension(t *testing.T) {
-	cert := pemToCert(t, testonly.PrecertPEMValid)
-	// Invalid because it's not asn.1 null
-	ext := pkix.Extension{Id: ctPoisonExtensionOID, Critical: false, Value: []byte{0x42, 0x42, 0x42}}
-
-	cert.Extensions = []pkix.Extension{ext}
-	_, err := IsPrecertificate(cert)
-
-	if err == nil {
-		t.Fatal("incorrectly accepted invalid CT extension")
-	}
-}
-
-func TestCertCheckerInvalidChainAccepted(t *testing.T) {
-	// This shouldn't validate as it's missing the intermediate cert
-	chainPem := []string{testonly.LeafSignedByFakeIntermediateCertPEM}
-	jsonChain := pemsToDERChain(t, chainPem)
-	trustedRoots := NewPEMCertPool()
-
-	if !trustedRoots.AppendCertsFromPEM([]byte(testonly.FakeCACertPEM)) {
-		t.Fatal("failed to load fake root")
-	}
-
-	_, err := ValidateChain(jsonChain, *trustedRoots, false)
-
-	if err == nil {
-		t.Fatal("verification accepted an invalid chain (missing intermediate)")
-	}
+	return cert
 }
 
-func TestCertCheckerInvalidChainRejectedOrdering(t *testing.T) {
-	// This chain shouldn't validate because the order of presentation is wrong
-	chainPem := []string{testonly.FakeIntermediateCertPEM, testonly.LeafSignedByFakeIntermediateCertPEM}
-	jsonChain := pemsToDERChain(t, chainPem)
-	trustedRoots := NewPEMCertPool()
-
-	if !trustedRoots.AppendCertsFromPEM([]byte(testonly.FakeCACertPEM)) {
-		t.Fatal("failed to load fake root")
-	}
-
-	_, err := ValidateChain(jsonChain, *trustedRoots, false)
-
-	if err == nil {
-		t.Fatal("verification accepted an invalid chain (ordering)")
-	}
+func makePoisonNonCritical(cert *x509.Certificate) *x509.Certificate {
+	// Poison extension needs to be marked as criticial to qualify as a pre-cert.
```

Comment:
Just for readability can you comment that it's creating an invalid pre-cert as in the method below. When I read "needs to be marked as be critical" and saw Critical:false  I had to think about it.